### DISCUSSION TITLE
Having a route that has no history

### DISCUSSION THREAD

**Discussion by gentledepp** [AvaloniaInside/Shell#61] [Q&A]
Hi! 
I went through the different navigation types, but could not find anything suitable.

Basically what we have is some route "authenticate" that can be called from any other route.
It is shown when the user has to login again (maybe the refresh-token was invalid or something)
Once logged in, I do not want to be able to "navigate back" to that view.

So I need a route that can be shown, but once the user leaves that page, does not create a history entry (speaking in browser language)

Is there any way I can achieve that?

Thanks!

**Comment by gentledepp** [comment]
@OmidID I guess my problem here is that I do not understand what the difference is between the navigationtypes:

```
public enum NavigateType
{
	Normal,
	ReplaceRoot,
	Modal,
	Replace,
	Top,
	Clear,
	Pop,
	HostedItemChange,
}

```

Could you please add documentation to the readme.md? 😇

**Reply by OmidID** [reply]
Hello @gentledepp 

Yes, exactly. You can find your answer in the NavigateType. 

Normal -> Add item in the navigation stack.
ReplaceRoot -> Clear entire stack and add current page to the root (History will be clear fully)
Modal -> It just open such as dialog (Maybe this one can help the case of login you have)
Replace -> Last page in the stack will remove and current page will add.
Top -> Find exists page in the stack and bring it to the top of the stack.
Clear -> Find exists page in stack and remove any top page until reach to found item.
Pop -> Do the back job
HostedItemChange -> Find the track page (Mainly use by item navigation or tab page change)

I will add it to the WIKI ;)

**Comment by gentledepp** [comment]
Thank you so much for the answer.
Can you maybe also add a scenario to the special navigation types like top and hosteditemchange? Just so people know when to use them 😀
But then I guess I messed something up. Because those navigate types do not change anything in my sample.

Also I do not know how to implement the login page now. Because it should not be kept on the navigatesrack. But I do not want to use Replace root since I want to keep the navigation history as such 🤔

Can I use "Modal" even if a page is not within some host?

**Reply by OmidID** [reply]
Hello @gentledepp You can take a looks to the example app and you can see welcome page get out from stack history once you click on login button. 

for the example of HostedItemChange i can mention tab change or left side menu when user click. It can handle the heretically host node change. 

Can you share an example app that i can check why you are blocked?

**Reply by gentledepp** [reply]
Thank you for your help!

I found out that I had an error when using the navigator.

When navigating from the welcome page to the main page, I thought I could specify the full path to the subroute of main ("/main/home") but that apparently does not work. I have to navigate using "/main".
Is there a reason for this?


The _actual_ problem is the following.
I created a "SplashView" that will be shown by default.
If the view determines there is some login to perform, it will forward the user to the "welcome" page, otherwise to the "main" page.

However, it seems I am navigating at the wrong spot:
![image](https://github.com/user-attachments/assets/97de025a-c3eb-4d68-8121-6fd7b0e48e94)

The result is, that it does navigate to the main view, but put the splash view as default subview of it (instead of home!!)
![mt-erber_deeplink](https://github.com/user-attachments/assets/f183a14d-1ed7-4945-b8f7-d6ae2fb79553)


So how can I achieve this?
My goal is to:
- use some callback of shell to notify the viewmodel that it can do its checks
- and let the viewmodel navigate without showing the current page to the user at all. :-D

I did a repro in my fork here
https://github.com/gentledepp/Shell/tree/repro/splashscreen

Any advise is much appreciated!

**Reply by gentledepp** [reply]
Hello @OmidID - just checking if you overlooked my answer? :-)
Can you give me any hint on where I could introduce such a check?

Like: App starts up, I need to verify if the user is logged in.
If (s)he is, I would like to navigate to the main view. Otherwise to the login view.

I thought I could do this in `InitializeAsync` but that apparently confuses the navigation :-(

**Reply by OmidID** [reply]
@gentledepp  I see in the video you post. you used modal for both splash and main page. 
If you look at the example for welcome page and home page you can see both are normal route. and then when you want to move from splash to main page you can use .NavigateAsync method and pass navigation type of Replace Root. This is the best way.

**Reply by OmidID** [reply]
@gentledepp By the way. you can create something like timeout in splash and then call the NavigateAsync.

**Reply by OmidID** [reply]
![image](https://github.com/user-attachments/assets/83701243-0f2e-4f8e-9a19-3fc2a671b332)
